### PR TITLE
Run scripts/PublicAPI/mark-shipped.cmd

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/37443. @JoeRobich fyi, this is almost certainly going to cause merge conflicts when it makes its way to master.

### CODE REVIEW

File: src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -0,0 +1 @@
...ptions.PerLanguageOption<bool>static Microsoft.CodeAnalysis.Options.Option<T>.implicit operator Microsoft.CodeAnalysis.Options.OptionKey(Microsoft.CodeAnalysis.Options.Option<T> option) -> Microsoft.CodeAnalysis.Options.OptionKeystatic Microsoft.CodeAnalysis.Options.OptionKey.operator !=(Microsoft.CodeAnalysis.Options.OptionKey left, Microsoft.CodeAnalysis.Options.OptionKey right) -> boolstatic Microsoft.CodeAnalysis.Options.OptionKey.operator ==(Microsoft.CodeAnalysis.Options.OptionKey left, Microsoft.CodeAnalysis.Options.OptionKey right) -> boolstatic readonly Microsoft.CodeAnalysis.CodeStyle.NotificationOption.Error -> Microsoft.CodeAnalysis.CodeStyle.NotificationOptionstatic readonly Microsoft.CodeAnalysis.CodeStyle.NotificationOption.None -> Microsoft.CodeAnalysis.CodeStyle.NotificationOptionstatic readonly Microsoft.CodeAnalysis.CodeStyle.NotificationOption.Silent -> Microsoft.CodeAnalysis.CodeStyle.NotificationOptionstatic readonly Microsoft.CodeAnalysis.CodeStyle.NotificationOption.Suggestion -> Microsoft.CodeAnalysis.CodeStyle.NotificationOptionstatic readonly Microsoft.CodeAnalysis.CodeStyle.NotificationOption.Warning -> Microsoft.CodeAnalysis.CodeStyle.NotificationOption
```
Comment (JoeRobich) [inline src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt]: This doesn't look right. one line with missing spaces.

File: src/Workspaces/Core/MSBuild/PublicAPI.Shipped.txt [hunk 2]
```diff
@@ -0,0 +1 @@
...s.Host.Mef.MSBuildMefHostServices.DefaultAssemblies.get -> System.Collections.Immutable.ImmutableArray<System.Reflection.Assembly>static Microsoft.CodeAnalysis.Host.Mef.MSBuildMefHostServices.DefaultServices.get -> Microsoft.CodeAnalysis.Host.Mef.MefHostServicesstatic Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.Create() -> Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspacestatic Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.Create(Microsoft.CodeAnalysis.Host.HostServices hostServices) -> Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspacestatic Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.Create(System.Collections.Generic.IDictionary<string, string> properties) -> Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspacestatic Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.Create(System.Collections.Generic.IDictionary<string, string> properties, Microsoft.CodeAnalysis.Host.HostServices hostServices) -> Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspacestatic Microsoft.CodeAnalysis.MSBuild.ProjectMap.Create() -> Microsoft.CodeAnalysis.MSBuild.ProjectMapstatic Microsoft.CodeAnalysis.MSBuild.ProjectMap.Create(Microsoft.CodeAnalysis.Solution solution) -> Microsoft.CodeAnalysis.MSBuild.ProjectMap
```
Comment (JoeRobich) [inline src/Workspaces/Core/MSBuild/PublicAPI.Shipped.txt]: This one looks problematic too

File: src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt (hunk 1 above)
Comment (333fred) [reply src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt]: Hmm. Our script might be messing up somehow.

File: src/CodeStyle/CSharp/Analyzers/PublicAPI.Shipped.txt [hunk 3]
```diff
@@ -0,0 +1,56 @@
+Microsoft.CodeAnalysis.CSharp.Formatting.BinaryOperatorSpacingOptions
```
Comment (jasonmalinowski) [inline src/CodeStyle/CSharp/Analyzers/PublicAPI.Shipped.txt]: @sharwell: what's the plan for a "public API" surface area in this assembly? Shouldn't this all be internal?

File: scripts/PublicApi/mark-shipped.ps1 [hunk 4]
```diff
@@ -7,6 +7,10 @@ $ErrorActionPreference = "Stop"
 function MarkShipped([string]$dir) {
     $shippedFilePath = Join-Path $dir "PublicAPI.Shipped.txt"
     $shipped = Get-Content $shippedFilePath
+    if ($null -eq $shipped) {
+        $shipped = @()
+    }
```
Comment (jaredpar) [inline scripts/PublicApi/mark-shipped.ps1]: ... how do you get into this case though. The `Get-Content` function should return `$null` only in error cases and that should throw.

File: src/CodeStyle/CSharp/Analyzers/PublicAPI.Shipped.txt (hunk 3 above)
Comment (jasonmalinowski) [reply src/CodeStyle/CSharp/Analyzers/PublicAPI.Shipped.txt]: (this is not a question about this particular thing but rather asking what it even means to run the analyzer on this assembly)

File: scripts/PublicApi/mark-shipped.ps1 (hunk 4 above)
Comment (333fred) [reply scripts/PublicApi/mark-shipped.ps1]: For new projects, such as CodeStyle. The PublicAPI.Shipped.txt files are empty, and `Get-Content` returns null.

Comment (jaredpar) [reply scripts/PublicApi/mark-shipped.ps1]: That's terrifying ...

File: src/CodeStyle/CSharp/Analyzers/PublicAPI.Shipped.txt (hunk 3 above)
Comment (sharwell) [reply src/CodeStyle/CSharp/Analyzers/PublicAPI.Shipped.txt]: The code style packages do not have a public API. Or more specifically, we always assume that there is no downstream code referencing these assemblies and are free to break them at will. We do not distribute the compiled binaries in any package which makes them available for reference/use by other code.

Comment (jasonmalinowski) [reply src/CodeStyle/CSharp/Analyzers/PublicAPI.Shipped.txt]: @sharwell: should we just turn off the analyzers for them then?

Comment (jaredpar) [reply src/CodeStyle/CSharp/Analyzers/PublicAPI.Shipped.txt]: If they don't have a public API that we intend to support then we should turn off the analyzer and delete the shipped / unshipped files. That is the expected state for non-public API assemblies.

Comment (sharwell) [reply src/CodeStyle/CSharp/Analyzers/PublicAPI.Shipped.txt]: That would be fine with me. However, if we plan to try and ship these later (so consolidate analyzer implementations in NuGet and VSIX) then we would need them added back.

**Pull request merged**